scripts/create_deprecated_module.lean

```diff
@@ -0,0 +1,175 @@
+  content it);
+* `keepTrailing` a boolean.
+
+It returns the content of `fileContent` up to the final import, including trailing whitespace and
+comments if `keepTrailing = true`.
+-/
+def getHeader (fname fileContent : String) (keepTrailing : Bool) : IO String := do
+  let (stx, _) ← Parser.parseHeader (Parser.mkInputContext fileContent fname)
+  let stx := if keepTrailing then stx.raw else stx.raw.unsetTrailing
+  let some substring := stx.getSubstring? | throw <| .userError "No substring: we have a problem!"
+  let upToAllImports : Substring := {substring with startPos := 0}
+  return upToAllImports.toString
+
+/--
+`getHeaderFromFileName fname keepTrailing` is similar to `getHeader`, except that it assumes that
+`fname` is the actual path of a file, and uses `fname`'s content as input to `getHeader`.
+-/
+def getHeaderFromFileName (fname : String) (keepTrailing : Bool) : IO String := do
+  getHeader fname (← IO.FS.readFile fname) keepTrailing
+
+/--
+`mkDeprecation customMessage` returns the formatted syntax
+`deprecated_module "customMessage" (since := "YYYY-MM-DD")`,
+where the message is the input and the date is today's date.
+-/
+def mkDeprecation (customMessage : String := "Auto-generated deprecation") :
+    CommandElabM Format := do
+  let msgStx := if customMessage.isEmpty then none else some <| Syntax.mkStrLit customMessage
+  let dateStx := Syntax.mkStrLit s!"{← Std.Time.PlainDate.now}"
+  let stx ← `(command|deprecated_module $[$msgStx]? (since := $dateStx))
+  liftCoreM <| PrettyPrinter.ppCategory `command stx
+
+/--
+The command `#create_deprecated_modules (n)? (comment)? (write)?` generates deprecated modules.
+
+Writing
+```lean
+#create_deprecated_modules 5 "These files are no longer relevant"
+```
+looks at the `lean` files in `Mathlib` that existed `4` commits ago
+(i.e. the commit that you see with `git log -5`) and that are no longer present.
+It shows how the corresponding deprecated modules should look like, using
+`"These files are no longer relevant"` as the (optional) comment.
+
+If the number of commits is not explicitly used, `#create_deprecated_modules` defaults to `2`,
+namely, the commit just prior to the current one.
+
+If the message is not explicitly used, `#create_deprecated_modules` defaults to
+`"Auto-generated deprecation"`.
+If you wish there to be no comment, use `#create_deprecated_modules 5 ""`.
+
+Note that the command applies the *same* comment to all the files that it generates.
+
+Finally, if everything looks correct, adding a final `write` actually generates the files:
+```lean
+#create_deprecated_modules 5 "These files are no longer relevant" write
+```
+-/
+syntax "#create_deprecated_modules" (ppSpace num)? (ppSpace str)? (&" write")? : command
+
+elab_rules : command
+| `(#create_deprecated_modules%$tk $[$nc:num]? $[$comment:str]? $[write%$write?]?) => do
+  let n := nc.getD (Syntax.mkNumLit "2") |>.getNat
+  let mut msgs := #[]
+  let getHash (n : Nat) := do
```

Comment:
How about a comment, say:
`--Get the hash of the commit at `git log -n` (and throw and error if that doesn't exist).`

Reply:
Comment added.